### PR TITLE
zcash_note_encryption: Note decryption fixes

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
utACK

File: components/zcash_note_encryption/src/lib.rs [hunk 1]
```diff
@@ -490,6 +512,14 @@ pub fn try_output_recovery_with_ock<D: Domain, Output: ShieldedOutput<D>>(
         domain.parse_note_plaintext_without_memo_ovk(&pk_d, &esk, output.epk(), &plaintext)?;
     let memo = domain.extract_memo(&plaintext);
 
+    // ZIP 212: Check that the esk provided to this function is consistent with the esk we
+    // can derive from the note.
```
Comment (daira) [inline components/zcash_note_encryption/src/lib.rs:516]: Does this have to be able to handle pre-ZIP212 plaintexts?

**Review by daira — APPROVED** [review]
utACK with a question.

File: components/zcash_note_encryption/src/lib.rs (hunk 1 above)
Comment (str4d) [reply components/zcash_note_encryption/src/lib.rs:516]: Pre-ZIP 212 note plaintexts will return `None` from `D::derive_esk()`.

**Review by ebfull — APPROVED** [review]

**Pull request merged**